worlds/LauncherComponents.py

```diff
@@ -100,10 +100,17 @@ def _install_apworld(apworld_src: str = "") -> Optional[Tuple[pathlib.Path, path
 
     apworld_path = pathlib.Path(apworld_src)
 
-    module_name = pathlib.Path(apworld_path.name).stem
+    apworld_name = apworld_path.name
     try:
         import zipfile
-        zipfile.ZipFile(apworld_path).open(module_name + "/__init__.py")
+        zip = zipfile.ZipFile(apworld_path)
+        directories = [f.filename.strip('/') for f in zip.filelist if f.CRC == 0 and f.file_size == 0 and f.filename.count('/') == 1]
+        if len(directories) == 1 and directories[0] in apworld_name:
```

Comment:
.name contains the extension, so using path.stem here would be safer (if the folder found was e.g. "ap", which is in "something.apworld")
```suggestion
        if len(directories) == 1 and directories[0] in apworld_path.stem:
```